mythril/laser/smt/array.py

```diff
@@ -26,9 +24,6 @@ def __getitem__(self, item: BitVec) -> BitVec:
 
     def __setitem__(self, key: BitVec, value: BitVec) -> None:
         """Sets an item in the array, key can be symbolic."""
-        if isinstance(value, Bool):
```

Comment:
If you move this logic to the storage object, then it should also be duplicated in memory right?

What is the benefit to moving it up to the datamodels?

Reply:
There was a problem with circular imports. So I had to move it. Memory currently uses Dict so there won’t be a problem with duplication for now